mne/viz/evoked.py

```diff
@@ -1723,7 +1755,42 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     axes.set_xlim(tmin, tmax)
 
     if len(conditions) > 1:
-        plt.legend(loc='best', ncol=1 + (len(conditions) // 5), frameon=True)
-
+        legend_params = dict(loc='best', frameon=True)
+        if split_legend:
+            if len(ls) > 1:
+                plt.legend(handles=ls, ncol=1 + (len(ls) // 4),
+                           **legend_params)
+        else:
+            plt.legend(ncol=1 + (len(conditions) // 5), **legend_params)
+
+    if sequential and split_legend:
+        import matplotlib.colors as mcolors
+        import matplotlib.cm
+        import matplotlib as mpl
+
+        def make_number(item):
+            return _strip_to_number(item.split("/")[0])
+
+        numbers = sorted([make_number(k) for k in colors_])
+        colors_l = list(colors_.keys())
+        colors_l.sort(key=make_number)
+        colors_m = [colors_[k] for k in colors_l]
+        l = plt.cm.jet.from_list(numbers + [numbers[-1] + 100],
+                                 colors_m + [colors_m[-1]])
+
+        norm = mpl.colors.BoundaryNorm(numbers + [numbers[-1] +1], l.N)
+        sm = matplotlib.cm.ScalarMappable(cmap=l, norm=norm
+                                         )
```

Comment:
That whole chunk of code is just very terrible in general ... I don't dare fixing it yet.